File: Dockerfile.sandbox.java
# goexec-sandbox-java — Java execution sandbox.
#
# Build:
#   docker build -f Dockerfile.sandbox.java -t goexec-sandbox-java:latest .

FROM eclipse-temurin:21-jdk

RUN useradd -ms /bin/sh sandbox
RUN mkdir /sandbox && chown sandbox:sandbox /sandbox

WORKDIR /sandbox
USER sandbox

CMD ["sh"]

File: Dockerfile.sandbox.go
FROM golang:1.21-alpine

RUN addgroup -S sandbox && adduser -S -G sandbox sandbox && \
    mkdir -p /sandbox /goexec-cache && \
    chown sandbox:sandbox /sandbox /goexec-cache

USER sandbox
ENV GOCACHE=/goexec-cache

# Pre-warm the Go build cache as the sandbox user.
# This compiles the stdlib and bakes the artifacts into the image layer,
# so the first real submission runs in <1s instead of 5-8s.
RUN cd /tmp && \
    printf 'package main\nimport "fmt"\nfunc main(){fmt.Println("ok")}' > warmup.go && \
    go run warmup.go && \
    rm warmup.go

WORKDIR /sandbox
CMD ["sh"]
